### DISCUSSION TITLE
Deprecated decorator on inherited classes

### DISCUSSION THREAD

**Discussion by aidandj** [microsoft/pyright#7273] [Q&A] [answered]
Should the deprecated decorator get picked up on an inherited class if an overridden method is marked deprecated in the base class? Or do the methods need to be marked deprecated in the child class as well.

```
from abc import abstractmethod
import typing as t
from typing_extensions import deprecated

class P(t.Protocol):
    @deprecated("testing")
    @abstractmethod
    def dep(self):
        ...

class Impl(P):
    def dep(self):
        ...

Impl().dep() # not marked deprecated

def test(p: P):
    p.dep() # marked deprecated as expected
```

**Comment by erictraut** [comment, marked as answer]
If a child overrides a method, it would need to be marked deprecated as well if that is your intent.

**Reply by aidandj** [reply]
Thanks Eric!
